timely/src/progress/subgraph.rs

```diff
@@ -463,22 +461,26 @@ where
         for ((location, time), diff) in self.pointstamp_tracker.pushed().drain() {
             // Targets are actionable, sources are not.
             if let crate::progress::Port::Target(port) = location.port {
-                if self.children[location.node].notify {
+                let child = &mut self.children[location.node];
+                if child.notify {
                     self.temp_active.push(Reverse(location.node));
                 }
                 // TODO: This logic could also be guarded by `.notify`, but
                 // we want to be a bit careful to make sure all related logic
                 // agrees with this (e.g. initialization, operator logic, etc.)
-                self.children[location.node]
-                    .shared_progress
-                    .borrow_mut()
-                    .frontiers[port]
-                    .update(time, diff);
+                if let Some(operator) = child.operator.as_mut() {
+                    operator.shared_progress_mut().frontiers[port].update(time, diff);
+                } else {
+                    // If the operator is closed and we are reporting progress at it, something has surely gone wrong.
+                    println!("Operator prematurely shut down: {}", child.name);
+                    println!("  {:?}", child.notify);
+                    // panic!();
```

Comment:
I was able to figure out what is going on here. This code patch definitely triggers even though an operator might be shut down. Ignoring it makes the tests pass but from the surrounding code and comments it sounds like that when an operator is shut down we should not push more progress info to it.

The test that makes this code path trigger is `dataflow::operators::enterleave::test::test_nested`

Reply:
I'll take a peek at this. The PR looks good other than commenting out the asserts, but it's great that you are observing them firing because .. should be immediately actionable!

Reply:
Oh great! Also, above I meant to write "I wasn't able to figure out". I've no idea why this change triggers it :)

Reply:
@frankmcsherry I debugged this today. In the subgraph handling we are creating a dummy operator as the 0th child that represent the external scope and we create a `PerOperatorState::empty()` to track it.

Previously that struct held onto a synthesized `Rc<RefCell<SharedProgress<T>>>` that didn't come from any operator and was fabricated on the spot. Since this PR moves the `SharedProgress` into the operators themelves this left these dummy operator states without a `SharedProgress` struct.

I fixed it by creating an `EmptyOperator` struct that implements `Operate` and `Schedule` that matches the properties that `PerOperatorState::empty()` already had, which now serves as the holder of `SharedProgress` for the representative.

I isolated the fix to the latest commit.